velox/exec/tests/ExpandTest.cpp

```diff
@@ -0,0 +1,180 @@
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<StringView>(
+              size,
+              [](auto row) {
+                auto str = std::string(row % 12, 'x');
+                return StringView(str);
+              }),
+      });
+
+  createDuckDbTable({data});
+
+  // Cube.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand({
+              {"k1", "k2", "a", "b", "0 as group_id_0"},
+              {"k1", "null as k2", "a", "b", "1 as group_id_0"},
+              {"null as k1", "k2", "a", "b", "2 as group_id_0"},
+              {"null as k1", "null as k2", "a", "b", "3 as group_id_0"},
+          })
+          .singleAggregation(
+              {"k1", "k2", "group_id_0"},
+              {"count(1) as count_1", "sum(a) as sum_a", "max(b) as max_b"})
+          .project({"k1", "k2", "count_1", "sum_a", "max_b"})
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY CUBE (k1, k2)");
+}
+
+TEST_F(ExpandTest, rollup) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"k1", "k2", "a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 11; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 17; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<StringView>(
+              size,
+              [](auto row) {
+                auto str = std::string(row % 12, 'x');
+                return StringView(str);
+              }),
+      });
+
+  createDuckDbTable({data});
+
+  // Rollup.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand(
+              {{"k1", "k2", "a", "b", "0 as group_id_0"},
+               {"k1", "null as k2", "a", "b", "1 as group_id_0"},
+               {"null as k1", "null as k2", "a", "b", "2 as group_id_0"}})
+          .singleAggregation(
+              {"k1", "k2", "group_id_0"},
+              {"count(1) as count_1", "sum(a) as sum_a", "max(b) as max_b"})
+          .project({"k1", "k2", "count_1", "sum_a", "max_b"})
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY ROLLUP (k1, k2)");
+}
+
+TEST_F(ExpandTest, countDistinct) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<StringView>(
+              size,
+              [](auto row) {
+                auto str = std::string(row % 12, 'x');
+                return StringView(str);
+              }),
+      });
+  createDuckDbTable({data});
+
+  // count distinct.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand(
+              {{"a", "null as b", "1 as group_id_0"},
```

Comment:
This plan doesn't look right as it doesn't use group_id_0 column generated by Expand. Would you please double check?

Reply:
@mbasmanova When I using the following sql, it throw exception. Do you have any input? Thanks.

SQL:
```
auto plan =
      PlanBuilder()
          .values({data})
          .expand(
              {{"a", "null as b", "1 as group_id_0"},
               {"null as b", "b", "2 as group_id_0"}})
          .singleAggregation(
              {"a", "b", "group_id_0"},
              {"count(distinct a) as count_a", "count(distinct b) as count_b"})
          .project({"count_a", "count_b"})
          .planNode();
```
Exception:
```
../../velox/exec/tests/utils/QueryAssertions.cpp:1058: Failure
Failed
Expected 1, got 1012
1012 extra rows, 1 missing rows
10 of extra rows:
        0 | 1
        0 | 1
        0 | 1
        0 | 1
        0 | 1
        0 | 1
        0 | 1
        0 | 1
        0 | 1
        0 | 1

1 of missing rows:
        1000 | 12
```